static_frame/core/util.py

```diff
@@ -3186,5 +3187,38 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
         raise IndexError(f'index {key} out of range for length {size} container.')
     return key % size
 
+def _unsized_children(obj):
+    # Check if iterable or a string first for fewer isinstance calls on common types
+    if hasattr(obj, '__iter__') and not isinstance(obj, str):
+        if (
+            (obj.__class__ is np.ndarray and obj.dtype.kind == DTYPE_OBJECT_KIND)
+            or isinstance(obj, tuple)
+            or isinstance(obj, list)
+            or isinstance(obj, set)
+            or isinstance(obj, frozenset)
```

Comment:
This can use `abc.Set`:
```
>>> from collections import abc                                                 
>>> isinstance(set(), abc.Set)                                                  
True
>>> isinstance(frozenset(), abc.Set)                                            
True
```